### DISCUSSION TITLE
502 Error on Dash, Hunt and Cases pages on new install

### DISCUSSION THREAD

**Discussion by Ataraxzy** [Security-Onion-Solutions/securityonion#10784] [2.4]
Left the server to run, untouched after installation, for a week or so, to work on getting a test box properly configured.

When I returned to it, I get a 502 error on the aforementioned pages.  Searches indicate that this is an index problem.  Okay, fine, I run the instructions found here: https://github.com/Security-Onion-Solutions/securityonion/discussions/9351

> so-elasticsearch-query [INDEX_NAME] -XDELETE

The problem is that all of the indices in question come from .ds-logs* and cannot be closed while running.  If I stop elasticsearch, the query/delete functionality is not there.

So, two questions:

1. What is causing this problem (again, this box is untouched after initial installation, there shouldn't be anything going wrong)?

2. How do I extricate myself from the problem without reinstalling it?

Thanks!

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/6271310/052eeb9e-833d-441c-a0c6-199123605f76)

**Comment by InfosecGoon** [comment]
When you're getting 502 errors there's usually some sort of issue with the Elasticsearch indices on the back end -- are there any errors in the logs in /opt/so/log/elasticsearch or /opt/so/log/soc? Does the node have adequate free disk space?

**Reply by Ataraxzy** [reply]
df -h shows that /nsm has 11%/35gb free.  It's pretty full, but 35gb is still quite a lot.  everything else has plenty of space, with the highest usage being 21%.

../log/soc/sync.log is empty of data
../log/elasticsearch/securityonion.log reports:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/6271310/945b731c-0594-4508-8666-8e86b5db2fd7)


Which, okay, that's fine, that should be a warning, not a stoppage.

**Reply by InfosecGoon** [reply]
So your total /nsm is about 350GB? How much data are you ingesting per day?

You can see your current indices and how large they are with "sudo so-elasticsearch-indices-list | sort".

**Reply by Ataraxzy** [reply]
Not certain, this is my first time setting up SO, and while it was installed on this box, I've been working at configuration on another box,  The indices are tiny.  Largest one is 23mb:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/6271310/d429babf-f322-4770-a5ff-ee89e564dd1e)

The 4 log indices show up red at the bottom.

**Reply by Ataraxzy** [reply]
Update:
Elastic fleet is showing the following errors:
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 8,
    "successful": 4,
    "skipped": 4,
    "failed": 4,
    "failures": [
      {
        "shard": 0,
        "index": ".ds-logs-elastic_agent-default-2023.07.08-000002",
        "node": null,
        "reason": {
          "type": "**no_shard_available_action_exception**",
          "reason": null
        }
      },
      {
        "shard": 0,
        "index": ".ds-logs-elastic_agent.filebeat-default-2023.07.08-000002",
        "node": null,
        "reason": {
          "type": "**no_shard_available_action_exception**",
          "reason": null
        }
      },
      {
        "shard": 0,
        "index": ".ds-logs-elastic_agent.fleet_server-default-2023.07.08-000002",
        "node": null,
        "reason": {
          "type": "**no_shard_available_action_exception**",
          "reason": null
        }
      },
      {
        "shard": 0,
        "index": ".ds-logs-elastic_agent.metricbeat-default-2023.07.08-000002",
        "node": null,
        "reason": {
          "type": "**no_shard_available_action_exception**",
          "reason": null
        }
      }
    ]
  },
  "hits": {
    "total": 0,
    "max_score": 0,
    "hits": []
  }
}
Edit: github doesn't like single CRs in text, and mashes everything, so I highlighted the error.
Edit2: it also doesn't like to format text inside code blocks... /sigh
Edit3: and it hates leading spaces when outside of code blocks. _seriously github? this is basic text formatting stuff_ whatever, it's (barely) readable.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/6271310/c651fc1e-fbdd-49e1-b53f-980653c6b25d)

**Reply by InfosecGoon** [reply]
If you do an "so-elasticsearch-query _cat/shards | wc -l", what is the output? Trying to figure out how many shards you have altogether.

**Reply by Ataraxzy** [reply]
it's a base Standalone install.

result of command:
42

**Reply by parsivalllll** [reply]
I'm facing the same issue but my result of command is 59 and it is also a standalone install

When I try entering hunt/dashboard/alerts its is showing this error "The Hunt interface may contain additional log messages to further diagnose the issue."

**Comment by dougburks** [comment]
> How do I extricate myself from the problem without reinstalling it?

@Ataraxzy Please keep in mind that if you're using Security Onion 2.4, it is considered BETA software and it is expected that you will reinstall it when newer versions are released.

**Comment by postrow** [comment]
Not to pile on here guys but I'm experiencing a similar "502" related issue on a base install of SO v2.4.10 (Hotfix: 20230821). This was pulled from logs within "/opt/so/log/elasticsearch".

[2023-10-03T23:59:07,620][WARN ][org.elasticsearch.cluster.routing.allocation.DiskThresholdMonitor] high disk watermark [85%] exceeded on [J9Dl1_55RK2BWQAEMQL3zw][securityonion][/usr/share/elasticsearch/data] free: 17.3gb[11%], shards will be relocated away from this node; currently relocating away shards totalling [0] bytes; the node is expected to continue to exceed the high disk watermark when these relocations are complete

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/a8764cdf-aae4-41c2-a89f-53c67d1eb1a3)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/e18d07b9-1504-4d65-8bd4-e78cefe78241)

![screenshot_1745](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/292fe4ec-12f8-4ccb-981b-cba6e845a670)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/f9c3b55b-55b8-440a-a962-87b86f2a1737)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/147430683/e4abc2aa-3ed7-43a4-a2be-5e4cc9503cbf)

**Reply by parsivalllll** [reply]
@postrow I'm facing this exact same issue from yesterday and I am not able to resolve it, Please let me if you managed to solve it and also how.
For the past hour, even the Kibana was not allowing me to log in.

**Comment by defensivedepth** [comment]
@postrow @parsivalllll Please start a new Discussion. This discussion references a much earlier version of 2.4 (beta) and is not relevant for the current version.
